plugins/europeana/utils.js

```diff
@@ -11,3 +11,61 @@ export function apiError(error) {
   apiError.statusCode = statusCode;
   return apiError;
 }
+
+const locales = require('../i18n/locales.json');
+
+function isEntity(value) {
+  return !!value && !!value.about;
+}
+
+function entityValues(values, locale) {
+  let entityValues = [];
+  for (const value of values || []) {
+    if (isEntity(value)) {
+      let entityValue = {};
+      if (value.prefLabel)  {
+        entityValue = langMapValueForLocale(value.prefLabel, locale);
+        entityValue['about'] = value.about;
+      } else {
+        entityValue = { code: '', values: [value.about], about: value.about };
+      }
+      entityValues = entityValues.concat(entityValue);
+    }
+  }
+  return entityValues;
+}
+
+/**
+ * Get the localised value for the current locale, with preferred fallbacks.
+ * Will return nothing if no value was found in any of the preferred locales.
+ * @param {Object} The LangMap
+ * @param {Object} locale current locale
+ * @return {{Object[]{language: String, values: Object[]}}} Language code and values, values may be strings or language maps themselves.
+ */
+export function langMapValueForLocale(langMap, locale) {
+  let languageKeys = ['en', 'eng', 'def', 'und']; // predefined set of preferred/fallback languages
+  const currentLocale = locales.find(l => l.code === locale);
+
+  if (currentLocale.code !== 'en') { // if language code is not en, add both 2 and 3 character language code to preferred languages
+    languageKeys.unshift(currentLocale.code);
+    languageKeys.unshift(currentLocale.isoAlpha3);
+  }
+
+  let returnVal = { values: [] };
+  for (let key of languageKeys) { // loop through all language key to find a match
```

Comment:
languageKeys can be a distinct function to simplify the logic above out of this function.